app_integrations/apps/salesforce.py

```diff
@@ -0,0 +1,395 @@
+
+        An example of log files response json body:
+        {
+            "totalSize": 2,
+            "done": True,
+            "records": [
+                {
+                    "attributes": {
+                        "type": "EventLogFile",
+                        "url": "/services/data/v32.0/sobjects/EventLogFile/0ATD00001bROAQ"
+                    },
+                    "Id": "0ATD000000001bROAQ",
+                    "EventType": "Console",
+                    "LogFile": "/services/data/v32.0/sobjects/EventLogFile/0ATD00001bROAQ/LogFile",
+                    "LogDate": "2014-03-14T00:00:00.000+0000",
+                    "LogFileLength": 2692.0
+                },
+                {
+                    "attributes": {
+                        "type": "EventLogFile",
+                        "url": "/services/data/v32.0/sobjects/EventLogFile/0ATD000000001SdOAI"
+                    },
+                    "Id": "0ATD000000001SdOAI",
+                    "EventType": "Console",
+                    "LogFile": "/services/data/v32.0/sobjects/EventLogFile/0ATD00001SdOAI/LogFile",
+                    "LogDate": "2014-03-13T00:00:00.000+0000",
+                    "LogFileLength": 1345.0
+                }
+            ]
+        }
+        """
+        log_files = []
+        for event_type in self._SALESFORCE_EVENT_TYPES:
+            event_type_format = 'AND+EventType+=+\'{}\''.format(event_type)
+            url = self._SALESFORCE_QUERY_URL.format(
+                instance_url=self._instance_url,
+                api_version=self._latest_api_version,
+                query=self._SALESFORCE_QUERY_FILTERS,
+                start_time=self._SALESFORCE_CREATE_AFTER.format(self._last_timestamp),
+                event_type=event_type_format
+            )
+            success, response = self._make_get_request(url, self._auth_headers)
+            if not success:
+                LOGGER.error('Failed to get a list of log files.')
+                return log_files
+
+            if response.get('records'):
+                log_files.extend([record['LogFile'] for record in response['records']
+                                  if record.get('LogFile')])
+
+        LOGGER.debug('Retrived %d log files', len(log_files))
+        return log_files
+
+    def _fetch_event_logs(self, log_file_path):
+        """Get event logs by sending GET request to each log file location.
+
+        Args:
+            log_file_path (str): log file location.
+
+        Returns:
+            list: a list of event logs or None.
+        """
+        url = '{}/{}'.format(self._instance_url, log_file_path)
+        data = None
+        try:
+            success, resp = self._make_get_request(url, self._auth_headers)
+            if not success:
+                LOGGER.error('Failed to get event logs')
+                return None
+
+            data = StringIO.StringIO(resp)
+            reader = csv.DictReader(data)
```

Comment:
isn't this going to be doing more work than we need to and increasing memory/size of logs we're sending to the rule processor? this could be problematic if we're running through thousands of lines and the rule processor already has logic for doing the csv -> json transformation. let's discuss this more if you'd like, but I'd be more comfortable if we just stripped off the header line (skip line 1) and send the lines as raw csv data. the rule processor will then use the schema to remap the line to it's header values and create the dict object

Reply:
Agreed.  The rule processor also has additional safety checks, etc for parsing out CSV data.  We also haven't really used it much yet and this would be a great opportunity to do so!

Reply:
Yes, I agree with you two. I will make the change. 😄